src/skadi/sample.cc

```diff
@@ -604,25 +605,30 @@ void sample::cache_initialisation(const std::string& data_source) {
 
   // messy but needed
   while (cache_->data_source.size() &&
-         cache_->data_source.back() == filesystem::path::preferred_separator) {
+         cache_->data_source.back() == std::filesystem::path::preferred_separator) {
     cache_->data_source.pop_back();
   }
+  cache_->cache.resize(TILE_COUNT);
+
+  const auto data_path = std::filesystem::path{cache_->data_source};
 
   // If data_source is empty, do not allocate/resize mapped cache.
-  if (cache_->data_source.empty()) {
+  if (cache_->data_source.empty() || !std::filesystem::is_directory(data_path)) {
     LOG_DEBUG("No elevation data_source was provided");
     return;
   }
-  cache_->cache.resize(TILE_COUNT);
 
   // check the directory for files that look like what we need
-  auto files = filesystem::get_files(cache_->data_source);
-  for (const auto& f : files) {
+  for (const auto& f : std::filesystem::recursive_directory_iterator(cache_->data_source)) {
+    if (!f.is_regular_file())
+      continue;
     // make sure its a valid index
-    auto data = cache_item_t::parse_hgt_name(f);
+    // TODO(nils): make this all based on filesystem::path instead of string
+    const auto fp_str = f.path().string();
+    auto data = cache_item_t::parse_hgt_name(fp_str);
     if (data && data->second != format_t::UNKNOWN) {
-      if (!cache_->insert(data->first, f, data->second)) {
-        LOG_WARN("Corrupt elevation data: " + f);
+      if (!cache_->insert(data->first, fp_str, data->second)) {
```

Comment:
on windows implicit conversion of filesystem::path to str doesn't work apparently, needed to do it explicitly